### DISCUSSION TITLE
Add 'Baseline' as an option for VerticalAlignment

### DISCUSSION THREAD

**Discussion by AndrewShepherd** [SixLabors/Fonts#287] [Ideas]
One of the many appeals of this library is that the details of how a font renders is abstracted away. (So far I have happily gotten away with not learning what a 'Glyph' is)

At the moment, the [VerticalAlignment](https://github.com/SixLabors/Fonts/blob/main/src/SixLabors.Fonts/VerticalAlignment.cs) enumeration consist of `Top`, `Center`, `Bottom`.

In comparison, the CSS 'vertical-align' property includes the value 'baseline'. (Which I think is the default)

This means that in HTML you can have flowing text where the style and the font size changes, but it all still reads like it is on the same line.

![i-am-feeling-perfectly-fine](https://user-images.githubusercontent.com/2449034/176796529-7c79dbcb-9c26-465f-a63f-2485d0e5aeba.png)

It would be a really good feature if we can choose `Baseline` as a vertical alignment option. 

(Just from reading [this discussion](https://github.com/SixLabors/Fonts/discussions/151) about font measurements I understand this might not be trivial)

**Comment by JimBobSquarePants** [comment]
Our [text runs](https://github.com/SixLabors/Fonts/pull/251) actually automatically align at the baseline. You can assign different font sizes to a individual runs.

This image from a [draft PR](https://github.com/SixLabors/ImageSharp.Drawing/pull/211) in the drawing repository indicates how the functionality will eventually allow rich text rendering.

![DrawRichTextWithMixOfPensAndBrushes_Solid500x200_(255,255,255,255)_RichText-F(40)](https://user-images.githubusercontent.com/166440/156829225-929014de-1826-44d5-bc7b-37399ba4e3ef.png)

Where a specific `Baseline` property might be useful is when rendering along a path. Currently `Bottom` includes the descender which means people have to offset the path if they want to avoid that. I'm not sure how to add that functionality as yet.
